### PR TITLE
Written a React recipe (with an example)

### PR DESCRIPTION
**Pull request by tomazzaman**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you add it to the readme: https://github.com/sindresorhus/ava#recipes

**Comment by vdemedes** [comment]
I think this recipe should focus strictly on React + AVA, without explaining version history, react code, etc. Examples of that:

- https://github.com/sindresorhus/ava/pull/462/files#diff-2cb79c7fb78b66228297358846395c3aR3
- https://github.com/sindresorhus/ava/pull/462/files#diff-2cb79c7fb78b66228297358846395c3aR21
- https://github.com/sindresorhus/ava/pull/462/files#diff-2cb79c7fb78b66228297358846395c3aR50

Could you please put more explanatory text in the recipe? I also noticed a few code suggestions, they should be listed below.

Thanks for your contribution and looking forward to merge this into master!

File: docs/recipes/react.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  /**
+   * Stateless components can't return null for now. React 0.15 is supposed to
+   * fix that, but for the time being use an empty <span> or <noscript> tag.
+   * https://github.com/facebook/react/issues/5355
+   */
+  return <span />;
+};
+
+FieldError.propTypes = {
```
Comment (vdemedes) [inline docs/recipes/react.md]: Not necessary for the demo

File: docs/recipes/react.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+And here is the test:
+
+```JavaScript
+import test from 'ava';
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import FieldError from '../../../js/components/form/FieldError.js';
+
+test('FieldError shows when the error is in array', assert => {
```
Comment (vdemedes) [inline docs/recipes/react.md]: `test('title', t => {` instead of `assert`

File: docs/recipes/react.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  const field = React.createElement(FieldError, {
+    errors,
+    attribute: 'email',
+    code: '123',
+  }, 'Error message');
+
+  const result = ReactDOMServer.renderToStaticMarkup(field);
+  const expectation = '<span class="form__error">Error message</span>';
+  assert.same(result, expectation);
```
Comment (vdemedes) [inline docs/recipes/react.md]: `t.is()` instead of `same()`

File: docs/recipes/react.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# React testing
+
+With 0.10.0, AVA stopped including your `.babelrc` (for reasons outlined in [this PR](https://github.com/sindresorhus/ava/pull/398)), which means you can't use JSX in your test/spec files. Note that this doesn't apply to your source files, they can still be written in JSX!
+
+The solution is just to use React's regular JS rendering methods, like `React.createElement`.
+
+So here's an example component we'd like to test:
+
+```JavaScript
```
Comment (kevva) [inline docs/recipes/react.md]: `JavaScript` => `js`

File: docs/recipes/react.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import React, { PropTypes } from 'react';
```
Comment (kevva) [inline docs/recipes/react.md]: I prefer `import {fn}` from 'module'` without spaces.

File: docs/recipes/react.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import { find } from 'lodash';
+
+const FieldError = ({ errors, attribute, code, children }) => {
```
Comment (kevva) [inline docs/recipes/react.md]: Same here, without spaces is nicer imo.

File: docs/recipes/react.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  code: PropTypes.string.isRequired,
+  children: PropTypes.string.isRequired,
+};
+
+export default FieldError;
+```
+
+And here is the test:
+
+```JavaScript
```
Comment (kevva) [inline docs/recipes/react.md]: `JavaScript` => `js`

File: docs/recipes/react.md (hunk 5 above)
Comment (sindresorhus) [inline docs/recipes/react.md]: :+1:

File: docs/recipes/react.md (hunk 6 above)
Comment (sindresorhus) [inline docs/recipes/react.md]: :+1:

**Comment by tomazzaman** [comment]
Sure, I'll make the necessary fixes. How does this work, do I close the PR, fix the code then resubmit?

**Comment by sindresorhus** [comment]
@tomazzaman No, you never need to resubmit the PR. Just push additional commits. See this guide: https://github.com/RichardLitt/docs/blob/master/amending-a-commit-guide.md

**Comment by tomazzaman** [comment]
Thanks for the guidance!

File: docs/recipes/react.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
+# React testing
+
+With 0.10.0, AVA stopped including your `.babelrc` (for reasons outlined in [this PR](https://github.com/sindresorhus/ava/pull/398)), which means you can't use JSX in your test/spec files. Note that this doesn't apply to your source files, they can still be written in JSX!
```
Comment (kentcdodds) [inline docs/recipes/react.md]: I was under the impression that there is soon to be a release that allows you to specify a `.babelrc` to use or some way to configure babel yourself. Is this not true?

Comment (jamestalmage) [inline docs/recipes/react.md]: I think we are pretty close to agreement on how it should look. But the implementation is not written yet.

Comment (kentcdodds) [inline docs/recipes/react.md]: In the interest of future-proofing this documentation, a reference to the relevant issue might be helpful here. Can't seem to find it right now though.

Comment (kentcdodds) [inline docs/recipes/react.md]: [Found it](https://github.com/sindresorhus/ava/issues/448)

Comment (jamestalmage) [inline docs/recipes/react.md]: #448

**Comment by sindresorhus** [comment]
@tomazzaman ping :)

**Comment by MoOx** [comment]
@tomazzaman Maybe we should use in the example `import { createRenderer } from "react-addons-test-utils"` with `const renderer = createRenderer(); renderer.render(...` ? It's the prefered way to test React components at the moment.

**Comment by sindresorhus** [comment]
@tomazzaman Would you be able to finish this? :)

**Comment by sindresorhus** [comment]
Closing for lack of activity. This is pretty moot now anyways as we finally have a solution for the Babel mess https://github.com/sindresorhus/ava/pull/573.

**Pull request closed without merging**